src/cli/utils/options.go

```diff
@@ -24,3 +27,12 @@ func Control() control.Options {
 
 	return opt
 }
+
+func InitConcurrencyControls(args []string) {
+	// FIXME: normally we want to control the limit according to the
+	// control options.  But we're racing for initialization here, so
+	// we'll set it to the default max as a starting point.
+	if slices.Contains(args, "exchange") {
+		graph.InitializeConcurrencyLimiter(4)
```

Comment:
We won't be able to change limiter capacity later, since this is a singleton and can only be initialized once?

Reply:
I've updated this to ensure we pass along the control options before initializing, so we shouldn't have any need to update the semaphore capacity afterward.